### PR TITLE
Simplify lroImpl

### PR DESCRIPTION
**Pull request by deyaaeldeen**
Simplify `lroImpl` class. Apparently, core v2 does the right thing and there is no need to do custom deserialization.

### CODE REVIEW

**Review by sarangan12 — APPROVED** [review]
I would highly recommend creating a detailed document explaining:

1. What is LRO? 
2. How was LRO handled in v4x?
3. What was the drawbacks?
4. How was LRO handled in V6 - Core V1?
5. How is LRO handled in V6 - Core V2?

with examples.

File: src/lroImpl.ts [hunk 1]
```diff
@@ -1,136 +1,18 @@
...
-    operationLocation: result.headers.get("operation-location"),
-    location: result.headers.get("location"),
-    requestMethod: result.request.method
-  };
-}
-
 export class LroImpl<T> implements LongRunningOperation<T> {
   constructor(
-    private sendOperationFn: SendOperationFn<T>,
+    private sendOperationFn: (args: any, spec: any) => Promise<LroResponse<T>>,
```
Comment (joheredi) [inline src/lroImpl.ts:5]: Is there a more specific type we can use instead of any?

**Review by joheredi — APPROVED** [review]
I love this change! Thanks for the hard work improving our LRO story!

File: src/lroImpl.ts (hunk 1 above)
Comment (deyaaeldeen) [reply src/lroImpl.ts:5]: Good point, I am addressing this in https://github.com/Azure/autorest.typescript/pull/1118.

**Comment by deyaaeldeen** [comment]
I also tested locally against the old testsuite that was using core v1 and it passes all the tests.

**Pull request merged**